File: GrowOne.Core/Settings/TemperatureWarningSettings.cs
/*
 * Copyright (c) 2022 Maximilian Bauer
 * Distributed under the GNU GPL v3. For full terms see the file "COPYING".
 */
#nullable enable

namespace GrowOne.Core.Settings
{
    public class TemperatureWarningSettings
    {
        public bool Enabled { get; set; }

        public float MinimumTemperature { get; set; }

        public float MaximumTemperature { get; set; }

        public TemperatureWarningSettings Clone()
        {
            return new TemperatureWarningSettings()
            {
                Enabled = Enabled,
                MinimumTemperature = MinimumTemperature,
                MaximumTemperature = MaximumTemperature
            };
        }
    }
}

File: GrowOne/Resources/WebResources/common/TemperatureWarningSettings.js
/*
 * Copyright (c) 2022 Maximilian Bauer
 * Distributed under the GNU GPL v3. For full terms see the file "COPYING".
 */
// @ts-check

import { Settings } from "./Settings.js";

export class TemperatureWarningSettings extends Settings {
   /** @type {boolean} */
   enabled;
   /** @type {number} */
   minimumTemperature;
   /** @type {number} */
   maximumTemperature;

   /**
    * @param {boolean} enabled
    * @param {number} minimumTemperature
    * @param {number} maximumTemperature
    */
   constructor(enabled, minimumTemperature, maximumTemperature) {
      super();
      this.enabled = enabled;
      this.minimumTemperature = minimumTemperature;
      this.maximumTemperature = maximumTemperature;
   }

   clone() {
      return new TemperatureWarningSettings(this.enabled, this.minimumTemperature, 
         this.maximumTemperature);
   }
}